src/parser/Expression.ts

```diff
@@ -358,27 +358,30 @@ export class FunctionParameterExpression extends Expression {
             result.push(' ');
             result.push(state.transpileToken(this.asToken));
             result.push(' ');
-            result.push(state.sourceNode(this.typeToken, this.type.toTypeString()));
+            result.push(state.sourceNode(this.typeToken!, this.type.toTypeString()));
         }
 
         return result;
     }
 
     public getTypedef(state: BrsTranspileState): TranspileResult {
-        return [
-            //name
-            this.name.text,
-            //default value
-            ...(this.defaultValue ? [
-                ' = ',
-                ...this.defaultValue.transpile(state)
-            ] : []),
-            //type declaration
-            ...(this.asToken ? [
-                ' as ',
-                this.typeToken?.text
-            ] : [])
-        ];
+        const results = [this.name.text] as TranspileResult;
+
+        if (this.defaultValue) {
+            results.push(' = ', ...this.defaultValue.transpile(state));
+        }
+
+        if (this.asToken) {
+            results.push(' as ');
+
+            // TODO: Is this conditional needed? Will typeToken always exist
+            // so long as `asToken` exists?
+            if (this.typeToken) {
+                results.push(this.typeToken.text);
+            }
+        }
+
+        return results;
```

Comment:
Of all my changes this seems the most likely to cause merge conflicts so I can back it out if needed.

Reply:
I think your changes make sense. They clean up this logic a good bit, so I welcome the merge conflicts!